### PR TITLE
identity-local-auth-magiclink

### PR DESCRIPTION
**Pull request by thukaraka**
Magic Link Authentication for Wso2 IS

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (IsuraD) [inline pom.xml:3]: change to 2021

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+~
```
Comment (sumedhe) [inline pom.xml:4]: Remove unnecessary new line

File: src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/JWTGenerator.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * JWT Generator related to magic link authentication flow.
+ */
+
```
Comment (sumedhe) [inline src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/JWTGenerator.java:16]: Remove unnecessary new line

File: src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/JWTGenerator.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+@SuppressWarnings("checkstyle:LineLength")
+public class JWTGenerator {
+
+    private final JWSAlgorithm signatureAlgorithm = new JWSAlgorithm(JWSAlgorithm.RS256.getName());
+
+    /**
```
Comment (sumedhe) [inline src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/JWTGenerator.java:22]: Add a comment line mentioning the functionality of this method.
Update in other places also.

File: src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/JWTGenerator.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+     * @param username       the username of the user
+     * @param sessionDataKey the sessionDataKey
+     * @param iss            the issuer of token
+     * @param aud            the audience of token
+     * @param exp            the expiration duration
```
Comment (sumedhe) [inline src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/JWTGenerator.java:27]: The parameter descriptions should start with capital and ends with fullstop.

File: src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticator.java [hunk 6]
```diff
@@ -0,0 +1,205 @@
+
```
Comment (sumedhe) [inline src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticator.java:1]: Remove unnecessary new line.

File: src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticator.java [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Authenticator of MagicLink.
+ */
+
```
Comment (sumedhe) [inline src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticator.java:49]: Remove unnecessary new line

File: src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticator.java [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+        Event identityMgtEvent = new Event(eventName, properties);
+        try {
+            MagicLinkServiceDataHolder.getInstance().getIdentityEventService().handleEvent(identityMgtEvent);
+        } catch (IdentityEventException var12) {
+            String errorMsg = "Error occurred while calling triggerNotification. " + var12.getMessage();
+            throw new AuthenticationFailedException(errorMsg, var12.getCause());
+        }
+    }
+
+}
```
Comment (sumedhe) [inline src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticator.java:205]: Add a newline in the end of the file.

File: src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticatorConstants.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (sumedhe) [inline src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticatorConstants.java:1]: Remove unnecessary new line

File: src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticatorConstants.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Constants related to magic link authentication flow.
+ */
+public abstract class MagicLinkAuthenticatorConstants {
+
+    public static final String AUTHENTICATOR_NAME = "MagicLinkAuthenticator";
+    public static final String AUTHENTICATOR_FRIENDLY_NAME = "MagicLink";
+    public static final String MAGICLINK_NOTIFICATION_PAGE = "authenticationendpoint/MagicLinkNotification.jsp";
+
+}
```
Comment (sumedhe) [inline src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/MagicLinkAuthenticatorConstants.java:30]: Add a newline at the end of the file.

**Comment by thukaraka** [comment]
repository deleted

**Pull request closed without merging**